python/ray/serve/config.py

```diff
@@ -84,6 +91,33 @@ def replicas_settings_valid(cls, max_replicas, values):
 
         return max_replicas
 
+    @validator("policy", always=True)
+    def serialize_policy(cls, policy, values):
+        """Serialize policy to be cloudpickled and base64 encoded utf8 string.
+
+        If policy is a string, it is possible to be already serialized when constructed
+        from protobuf. In this case, we just return the already serialized string. Else
+        we import the policy and serialize it before return.
```

Comment:
Why do we need to base64 encode the string? It seems very unlikely to me that this is actually needed. If it is, it should be clearly documented why we need this extra level of encoding

And why do we need this dynamic/branching behavior? Instead we can just have the constructor always take in a serialized policy

Reply:
This is likely the same reason as https://github.com/ray-project/ray/blob/bbbe66200eafda8fd0be62aafff3ba40c071bb21/python/ray/serve/_private/config.py#L222. Somehow if we just pass in a raw string, then it's getting base64 encoded. But likely will not be needed after redesign to ensure runtime_env works.